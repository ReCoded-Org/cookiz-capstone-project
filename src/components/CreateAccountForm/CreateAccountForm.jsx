import Image from "next/image";
import { useState } from "react";

export default function CreateAccountForm() {
    const [firstname, setFirstname] = useState("");
    const [lastname, setLastname] = useState("");
    const [password, setPassword] = useState("");
    const [email, setEmail] = useState("");
    const [phone, setPhone] = useState("");
    const [address, setAddress] = useState("");
    const [city, setCity] = useState("");
    const [county, setCounty] = useState("");

    /* Debugging */
    // console.log(
    //     firstname,
    //     lastname,
    //     password,
    //     email,
    //     phone,
    //     address,
    //     city,
    //     county
    // );

    return (
        // component container
        <div className='flex flex-row flex-wrap justify-center'>
            {/* form container */}
            <div className='flex flex-col space-y-4 px-10 pt-9 pb-9'>
                <h1 className='text-center text-2xl font-bold text-primary'>
                    Create An Account
                </h1>
                <form className='flex w-96 flex-col space-y-5'>
                    {/* first name */}
                    <input
                        className='w-full appearance-none rounded border py-2 px-3 text-sm font-bold leading-tight text-gray-700 shadow'
                        id='firstName'
                        type='text'
                        placeholder='First Name'
                        value={firstname}
                        onChange={(event) => setFirstname(event.target.value)}
                    />
                    {/* last name */}
                    <input
                        className='w-full appearance-none rounded border py-2 px-3 text-sm font-bold leading-tight text-gray-700 shadow focus:outline-none'
                        id='lastName'
                        type='text'
                        placeholder='Last Name'
                        value={lastname}
                        onChange={(event) => setLastname(event.target.value)}
                    />
                    {/* password */}
                    <input
                        className='w-full appearance-none rounded border py-2 px-3 text-sm font-bold leading-tight text-gray-700 shadow focus:outline-none'
                        id='password'
                        type='password'
                        placeholder='Password'
                        value={password}
                        onChange={(event) => setPassword(event.target.value)}
                    />
                    {/* email */}
                    <input
                        className='w-full appearance-none rounded border py-2 px-3 text-sm font-bold leading-tight text-gray-700 shadow focus:outline-none'
                        id='email'
                        type='text'
                        placeholder='Email'
                        value={email}
                        onChange={(event) => setEmail(event.target.value)}
                    />
                    {/* phone number */}
                    <input
                        className='w-full appearance-none rounded border py-2 px-3 text-sm font-bold leading-tight text-gray-700 shadow focus:outline-none'
                        id='phoneNumber'
                        type='text'
                        placeholder='Phone Number'
                        value={phone}
                        onChange={(event) => setPhone(event.target.value)}
                    />
                    {/* Address */}
                    <input
                        className='w-full appearance-none rounded border py-2 px-3 text-sm font-bold leading-tight text-gray-700 shadow focus:outline-none'
                        id='address'
                        type='text'
                        placeholder='Address'
                        value={address}
                        onChange={(event) => setAddress(event.target.value)}
                    />
                    {/* City */}
                    {/* TODO: add the ZIP CODE field */}
                    <input
                        className='w-full appearance-none rounded border py-2 px-3 text-sm font-bold leading-tight text-gray-700 shadow focus:outline-none'
                        id='city'
                        type='text'
                        placeholder='City'
                        value={city}
                        onChange={(event) => setCity(event.target.value)}
                    />
                    {/* County */}
                    <input
                        className='w-full appearance-none rounded border py-2 px-3 text-sm font-bold leading-tight text-gray-700 shadow focus:outline-none'
                        id='county'
                        type='text'
                        placeholder='Country'
                        value={county}
                        onChange={(event) => setCounty(event.target.value)}
                    />
                    {/* Terms and Conditions */}
                    <label className='block w-full items-center font-bold text-black'>
                        <input
                            className='mr-2 rounded bg-gray-100 text-primary focus:ring-primary'
                            id='termsAndCoditions-checkbox'
                            type='checkbox'
                        />
                        <span className='text-sm'>
                            I agree to Terms and Conditions
                        </span>
                    </label>
                    {/* submit button */}
                    <button className='focus:shadow-outline rounded-xl bg-primary py-2  px-4 font-bold text-white shadow hover:bg-emerald-700'>
                        Register
                    </button>
                    {/* cancel button */}
                    <button className='focus:shadow-outline rounded-xl border border-primary  bg-transparent py-2 px-4 font-bold text-primary shadow hover:bg-primary  hover:text-white'>
                        Cancel
                    </button>
                    {/* sign in redirecting */}
                    <div className='text-center text-xs font-bold text-black'>
                        <span>Already Have An Account?</span>{" "}
                        <span className='text-blue-500 underline'>
                            <a href=''>Sign in</a>
                        </span>
                    </div>
                </form>
            </div>

            {/* image section */}
            <div className='flex max-w-screen-sm items-center bg-signUpImageColor'>
                <Image
                    src='/images/signup.png'
                    alt='test'
                    width={500}
                    height={500}
                ></Image>
            </div>
        </div>
    );
}
