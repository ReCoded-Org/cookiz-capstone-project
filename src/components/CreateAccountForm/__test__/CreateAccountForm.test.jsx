import renderer from "react-test-renderer";

import CreateAccountForm from "../CreateAccountForm";

// eslint-disable-next-line
it("Create Account Form Component", () => {
    const tree = renderer.create(<CreateAccountForm />).toJSON();
    // eslint-disable-next-line no-undef
    expect(tree).toMatchSnapshot();
});
