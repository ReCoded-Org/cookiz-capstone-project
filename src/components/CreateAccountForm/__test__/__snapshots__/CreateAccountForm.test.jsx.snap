// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Account Form Component 1`] = `
<div
  className="flex flex-row flex-wrap justify-center"
>
  <div
    className="flex flex-col space-y-4 px-10 pt-9 pb-9"
  >
    <h1
      className="text-center text-2xl font-bold text-primary"
    >
      Create An Account
    </h1>
    <form
      className="flex w-96 flex-col space-y-5"
    >
      <input
        className="w-full appearance-none rounded border py-2 px-3 text-sm font-bold leading-tight text-gray-700 shadow"
        id="firstName"
        onChange={[Function]}
        placeholder="First Name"
        type="text"
        value=""
      />
      <input
        className="w-full appearance-none rounded border py-2 px-3 text-sm font-bold leading-tight text-gray-700 shadow focus:outline-none"
        id="lastName"
        onChange={[Function]}
        placeholder="Last Name"
        type="text"
        value=""
      />
      <input
        className="w-full appearance-none rounded border py-2 px-3 text-sm font-bold leading-tight text-gray-700 shadow focus:outline-none"
        id="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <input
        className="w-full appearance-none rounded border py-2 px-3 text-sm font-bold leading-tight text-gray-700 shadow focus:outline-none"
        id="email"
        onChange={[Function]}
        placeholder="Email"
        type="text"
        value=""
      />
      <input
        className="w-full appearance-none rounded border py-2 px-3 text-sm font-bold leading-tight text-gray-700 shadow focus:outline-none"
        id="phoneNumber"
        onChange={[Function]}
        placeholder="Phone Number"
        type="text"
        value=""
      />
      <input
        className="w-full appearance-none rounded border py-2 px-3 text-sm font-bold leading-tight text-gray-700 shadow focus:outline-none"
        id="address"
        onChange={[Function]}
        placeholder="Address"
        type="text"
        value=""
      />
      <input
        className="w-full appearance-none rounded border py-2 px-3 text-sm font-bold leading-tight text-gray-700 shadow focus:outline-none"
        id="city"
        onChange={[Function]}
        placeholder="City"
        type="text"
        value=""
      />
      <input
        className="w-full appearance-none rounded border py-2 px-3 text-sm font-bold leading-tight text-gray-700 shadow focus:outline-none"
        id="county"
        onChange={[Function]}
        placeholder="Country"
        type="text"
        value=""
      />
      <label
        className="block w-full items-center font-bold text-black"
      >
        <input
          className="mr-2 rounded bg-gray-100 text-primary focus:ring-primary"
          id="termsAndCoditions-checkbox"
          type="checkbox"
        />
        <span
          className="text-sm"
        >
          I agree to Terms and Conditions
        </span>
      </label>
      <button
        className="focus:shadow-outline rounded-xl bg-primary py-2 px-4 font-bold text-white shadow hover:bg-emerald-700"
      >
        Register
      </button>
      <button
        className="focus:shadow-outline rounded-xl border border-primary bg-transparent py-2 px-4 font-bold text-primary shadow hover:bg-primary hover:text-white"
      >
        Cancel
      </button>
      <div
        className="text-center text-xs font-bold text-black"
      >
        <span>
          Already Have An Account?
        </span>
         
        <span
          className="text-blue-500 underline"
        >
          <a
            href=""
          >
            Sign in
          </a>
        </span>
      </div>
    </form>
  </div>
  <div
    className="max-w-screen-sm flex items-center bg-signUpImageColor"
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27500%27%20height=%27500%27/%3e"
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        alt="test"
        data-nimg="intrinsic"
        decoding="async"
        onError={[Function]}
        onLoad={[Function]}
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="test"
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Fsignup.png&w=1080&q=75"
          srcSet="/_next/image?url=%2Fimages%2Fsignup.png&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fsignup.png&w=1080&q=75 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </div>
</div>
`;
